plugins/htpasswd/src/crypt3.ts

```diff
@@ -16,8 +16,12 @@ import crypt from 'unix-crypt-td-js';
  * distros), sha256 or sha512. Default is sha512.
  * @returns {string} Generated salt string
  */
-export function createSalt(type = 'sha512'): string {
+export function createSalt(type = 'crypt'): string {
```

Comment:
If we change this, don't we invalidate tokens already generated? 🤔 or I missing something.

Reply:
No, createSalt is only used to generate new salts, i.e. for encrypting new passwords. When validating existing ones, the salt is taken from the existing entry itself. In the case of the change password function, it in fact didn't do that and therefore broke, that's what the second commit addresses.

The change in default here simply means that variable salts will be generated going forward, i.e. for new users or when users change their password.

Reply:
I think you mean hashing. =)